[MathML] Support CSS size containment with contain-intrinsic-size properties

Ahmad-S792 opened the pull request:
#### 34374f3a123438b30056008d0ba09009a29e1dda
<pre>
[MathML] Support CSS size containment with contain-intrinsic-size properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=304018">https://bugs.webkit.org/show_bug.cgi?id=304018</a>
<a href="https://rdar.apple.com/166323213">rdar://166323213</a>

Reviewed by Alan Baradlay.

Implement support for CSS Containment&apos;s size containment in MathML
elements. When contain: size or contain: inline-size is specified along
with contain-intrinsic-inline-size or contain-intrinsic-block-size, use
these explicit intrinsic sizes instead of computing from content.

This enables MathML elements to participate in size containment
optimizations, matching the behavior of other CSS layout models.

Specification:
<a href="https://drafts.csswg.org/css-contain-2/#propdef-contain-intrinsic-size">https://drafts.csswg.org/css-contain-2/#propdef-contain-intrinsic-size</a>
<a href="https://drafts.csswg.org/css-contain-2/#containment-size">https://drafts.csswg.org/css-contain-2/#containment-size</a>

* Source/WebCore/rendering/mathml/RenderMathMLBlock.cpp:
(WebCore::RenderMathMLBlock::sizeAppliedToMathContent):
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Progression

&gt; Remove Platform Specific Expectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Removed.

Canonical link: <a href="https://commits.webkit.org/304458@main">https://commits.webkit.org/304458@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/41c8a9b2ef55fd3cf7125c4b64540ec90523f162

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135285 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7670 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46588 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/142790 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/87030 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/563510a1-5e23-41f8-8c2e-4fe53d694f66) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137154 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8307 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7518 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103380 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/70737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/4c98203c-036f-4c22-988c-f6de61a45d5c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138231 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5935 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121277 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84243 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/6a3e47ff-05b4-470c-bac1-882a5b1df3a8) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5721 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3325 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3377 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/114926 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39459 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145486 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7353 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40029 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111759 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7396 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6160 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112126 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28509 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5572 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117577 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61282 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7407 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35702 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7160 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/70954 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7382 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7263 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

fred-wang reviewed (commented):
Will defer to Alan as I'm not very familiar how size containment is implemented in WebKit. But the patch looks good to me and it's nice we are passing tests for all MathML elements.

Ahmad-S792 committed:
[MathML] Support CSS size containment with contain-intrinsic-size properties
https://bugs.webkit.org/show_bug.cgi?id=304018
rdar://166323213

Reviewed by Alan Baradlay.

Implement support for CSS Containment's size containment in MathML
elements. When contain: size or contain: inline-size is specified along
with contain-intrinsic-inline-size or contain-intrinsic-block-size, use
these explicit intrinsic sizes instead of computing from content.

This enables MathML elements to participate in size containment
optimizations, matching the behavior of other CSS layout models.

Specification:
https://drafts.csswg.org/css-contain-2/#propdef-contain-intrinsic-size
https://drafts.csswg.org/css-contain-2/#containment-size

* Source/WebCore/rendering/mathml/RenderMathMLBlock.cpp:
(WebCore::RenderMathMLBlock::sizeAppliedToMathContent):
* LayoutTests/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Progression

> Remove Platform Specific Expectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Removed.
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/mathml/relations/css-styling/size-containment-001.tentative-expected.txt: Removed.

Canonical link: https://commits.webkit.org/304458@main
